core/src/banking_stage/immutable_deserialized_packet.rs

```diff
@@ -116,22 +119,37 @@ impl ImmutableDeserializedPacket {
         votes_only: bool,
         address_loader: impl AddressLoader,
         reserved_account_keys: &HashSet<Pubkey>,
-    ) -> Option<SanitizedTransaction> {
+    ) -> Option<RuntimeTransaction<SanitizedTransaction>> {
         if votes_only && !self.is_simple_vote() {
             return None;
         }
-        let tx = SanitizedTransaction::try_new(
-            self.transaction().clone(),
-            *self.message_hash(),
-            self.is_simple_vote(),
-            address_loader,
-            reserved_account_keys,
+
+        let tx = RuntimeTransaction::<SanitizedVersionedTransaction>::try_from(
+            self.transaction.clone(),
+            Some(self.message_hash),
+            Some(self.is_simple_vote),
         )
+        .and_then(|tx| {
+            RuntimeTransaction::<SanitizedTransaction>::try_from(
+                tx,
+                address_loader,
+                reserved_account_keys,
+            )
+        })
         .ok()?;
+
         Some(tx)
     }
 }
 
+// Eq and PartialEq MUST be consistent with PartialOrd and Ord
```

Comment:
Yeah this can be separated out, did't actually have much to do with this change.
It's still a bug and should be fixed - see description of #3098

Reply:
This PR was seprated out. Now this PR just moves the derived Eq to below since derive doesn't work if inner fields don't implement Eq (even though the partialEq is not derived...rust is weird sometimes)